torch/_inductor/codegen/common.py

```diff
@@ -833,6 +841,35 @@ def reduction(name, dtype, src_dtype, reduction_type, index, value):
                     name, dtype, src_dtype, reduction_type, index, value
                 )
 
+            @staticmethod
+            def bucketize(
+                values,
+                offsets_name: str,
+                offsets_size,
+                indexing_dtype: str,
+                right: bool,
+            ):
+                """
+                [Note: Inductor bucketize op]
+
+                Given values (tensor) and offsets_name (reference to the name of a 1D
+                tensor), calculate the bucket that each value belongs to.
+
+                e.g. for values [-1, 0, 1, 2, 3, 4, 5, 9], offsets [0, 4, 8], right=False
+                return =        [ 0, 1, 1, 1, 1, 2, 2, 3].
```

Comment:
Although this, indeed, seems to correspond to the semantics of `torch.bucketize`, in the jagged tensor context, seems that we'll need to subtract 1 from the result (as the values falling into the first bucket should have index 0).

Reply:
In some rough benchmarks I didn't see any measurable difference in perf from subtracting the 1 vs. not having to do that (in fbgemm lowerings)